assets/components/odf-lvm/topolvm-node_daemonset.yaml

```diff
@@ -12,7 +12,7 @@ spec:
   template:
     metadata:
       annotations:
-        odf-lvm.microshift.io/lvmd_config_sha256sum: "{{ Sha256sum .Config }}"
+        odf-lvm.microshift.io/lvmd_config_sha256sum: "{{ if .lvmd }}{{ Sha256sum .lvmd }}{{ end }}" # do not generate sha's for 0 length strings
```

Comment:
Is the comment always going to end up in the output? Is there a template language syntax for a comment that would strip it when the template is rendered?

Would this be clearer if we put the logic in the template data passed in to the render call, instead of the template? For example, an `.lvmd` field and a separate `.lvmdChecksum` field?

Reply:
In https://github.com/openshift/microshift/blob/main/pkg/components/render.go#L41, could you please try adding `template.New("").Funcs(templateFuncs).Option("missingkey=error")`? I suspect you're missing the key and then the correct way to write something conditionally is to use `{{if ne (index .YourStruct "your_field") ""}}`. `{{if}}` cannot deal with `nil`, but `index` can and will return an empty string if `nil`.

Reply:
added, thanks!